browser/src/canvas/sections/URLPopUpSection.ts

```diff
@@ -0,0 +1,112 @@
+
+class URLPopUpSection extends HTMLObjectSection {
+    static sectionName = 'URL PopUp';
+	containerId = 'hyperlink-pop-up-preview';
+	linkId = 'hyperlink-pop-up';
+	copyButtonId = 'hyperlink-pop-up-copy';
+	editButtonId = 'hyperlink-pop-up-edit';
+	removeButtonId = 'hyperlink-pop-up-remove';
+
+    constructor (url: string, documentPosition: cool.SimplePoint) {
+        super(URLPopUpSection.sectionName, null, null, documentPosition);
+
+		const objectDiv = this.getHTMLObject();
+		objectDiv.remove();
+		document.getElementById('document-container').appendChild(objectDiv);
+
+		objectDiv.style.backgroundColor = 'floralwhite';
+		objectDiv.style.borderRadius = '5px';
+		objectDiv.style.border = '1px solid black';
+
+		this.sectionProperties.url = url;
+
+		this.createUIElements(url);
+		this.setUpCallbacks();
+
+		app.map.hyperlinkPopup = this;
+
+		document.getElementById('hyperlink-pop-up').title = url;
+
+		if (app.map['wopi'].EnableRemoteLinkPicker)
+			app.map.fire('postMessage', { msgId: 'Action_GetLinkPreview', args: { url: url } });
+    }
+
+	createUIElements(url: string) {
+		const parent = this.getHTMLObject();
+		L.DomUtil.createWithId('div', this.containerId, parent);
+
+        var link = L.DomUtil.createWithId('a', this.linkId, parent);
+		link.innerText = url;
+
+		const copyBtn = L.DomUtil.createWithId('div', this.copyButtonId, parent);
+		L.DomUtil.addClass(copyBtn, 'hyperlink-popup-btn');
+		copyBtn.setAttribute('title', _('Copy link location'));
+
+        const imgCopyBtn = L.DomUtil.create('img', 'hyperlink-pop-up-copyimg', copyBtn);
+		L.LOUtil.setImage(imgCopyBtn, 'lc_copyhyperlinklocation.svg', app.map);
+		imgCopyBtn.setAttribute('width', 18);
+		imgCopyBtn.setAttribute('height', 18);
+		imgCopyBtn.setAttribute('style', 'padding: 4px');
+
+		const editBtn = L.DomUtil.createWithId('div', this.editButtonId, parent);
+		L.DomUtil.addClass(editBtn, 'hyperlink-popup-btn');
+		editBtn.setAttribute('title', _('Edit link'));
+
+		const imgEditBtn = L.DomUtil.create('img', 'hyperlink-pop-up-editimg', editBtn);
+		L.LOUtil.setImage(imgEditBtn, 'lc_edithyperlink.svg', app.map);
+		imgEditBtn.setAttribute('width', 18);
+		imgEditBtn.setAttribute('height', 18);
+		imgEditBtn.setAttribute('style', 'padding: 4px');
+
+		const removeBtn = L.DomUtil.createWithId('div', this.removeButtonId, parent);
+		L.DomUtil.addClass(removeBtn, 'hyperlink-popup-btn');
+		removeBtn.setAttribute('title', _('Remove link'));
+
+		const imgRemoveBtn = L.DomUtil.create('img', 'hyperlink-pop-up-removeimg', removeBtn);
+		L.LOUtil.setImage(imgRemoveBtn, 'lc_removehyperlink.svg', app.map);
+		imgRemoveBtn.setAttribute('width', 18);
+		imgRemoveBtn.setAttribute('height', 18);
+		imgRemoveBtn.setAttribute('style', 'padding: 4px');
+	}
+
+	setUpCallbacks() {
+		document.getElementById(this.linkId).onclick = function() {
```

Comment:
please use arrow functions in the new code, then `this` should be already the right thing